perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -3704,7 +3703,31 @@ public void updateExistingMemberWhileSynchronization(PerunSession sess, Group gr
 		addUserExtSources(sess, candidate, memberToUpdate);
 
 		//Set correct member Status
-		updateMemberStatus(sess, memberToUpdate);
+		try {
+			if (isMemberStatusChangeAllowed(sess, group)) {
+				updateMemberStatus(sess, memberToUpdate);
+			}
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException ex) {
+			throw new InternalErrorException(ex);
+		}
+	}
+
+	/**
+	 * Return false if attribute group:keepMemberStatus is set to true.
+	 * True otherwise.
+	 *
+	 * @param sess session
+	 * @param group group to be synchronized
+	 *
+	 * @return false if change of member's status is not allowed, true otherwise
+	 *
+	 * @throws InternalErrorException if obtaining attribute failed
+	 * @throws WrongAttributeAssignmentException if incorrect assignment of attribute
+	 * @throws AttributeNotExistsException if attribute does not exist in database
+	 */
+	private boolean isMemberStatusChangeAllowed(PerunSession sess, Group group) throws WrongAttributeAssignmentException, AttributeNotExistsException {
```

Comment:
Should this method return the value of the new attribute? Seems to me that it only returns true when the attribute is not set, which doesn't make much sense. The javadocs are really confusing as well.